src/modules/CSADigitizer/CSADigitizerModule.cpp

```diff
@@ -0,0 +1,337 @@
+                                       .c_str());
+            getROOTDirectory()->WriteTObject(charge_graph, name.c_str());
+
+	    // -------- now the same for the amplified (and shaped) pulses
+            std::vector<double> amptime(output_vec.size());
+            // clang-format off
+            std::generate(amptime.begin(), amptime.end(), [n = 0.0, timestep]() mutable {  auto now = n; n += timestep; return now; });
+            // clang-format on
+
+	    // asv there needs to be a better way to plot in mV ?
+            std::vector<double> output_in_mV(output_vec.size());
+	    double scaleConvert{1e9};
+	    std::transform(output_vec.begin(), output_vec.end(), output_in_mV.begin(), [&scaleConvert](auto& c){return c*scaleConvert;});
+
+            name = "output_ev" + std::to_string(event_num) + "_px" + std::to_string(pixel_index.x()) + "-" +
+	           std::to_string(pixel_index.y());
+            auto output_graph = new TGraph(static_cast<int>(output_vec.size()), &amptime[0], &output_in_mV[0]);
+            output_graph->GetXaxis()->SetTitle("t [ns]");
+            output_graph->GetYaxis()->SetTitle("CSA output [mV]");
+            output_graph->SetTitle(("Amplifier signal in pixel (" + std::to_string(pixel_index.x()) + "," +
+                                   std::to_string(pixel_index.y()) +  ")")
+                                      .c_str());
+            getROOTDirectory()->WriteTObject(output_graph, name.c_str());
+
+
+	    // -------- now the same for the amplified (and shaped) pulses with noise
+	    // asv there needs to be a better way to plot in mV ?
+            std::vector<double> output_with_noise_in_mV(output_with_noise.size());
+	    std::transform(output_with_noise.begin(), output_with_noise.end(), output_with_noise_in_mV.begin(), [&scaleConvert](auto& c){return c*scaleConvert;});
+
+            name = "output_with_noise_ev" + std::to_string(event_num) + "_px" + std::to_string(pixel_index.x()) + "-" +
+	           std::to_string(pixel_index.y());
+            auto output_with_noise_graph = new TGraph(static_cast<int>(output_with_noise.size()), &amptime[0], &output_with_noise_in_mV[0]);
+            output_with_noise_graph->GetXaxis()->SetTitle("t [ns]");
+            output_with_noise_graph->GetYaxis()->SetTitle("CSA output [mV]");
+            output_with_noise_graph->SetTitle(("Amplifier signal with added noise in pixel (" + std::to_string(pixel_index.x()) + "," +
+                                   std::to_string(pixel_index.y()) +  ")")
+                                      .c_str());
+            getROOTDirectory()->WriteTObject(output_with_noise_graph, name.c_str());
+
+	}
+
+
+	// asv output "charge" of output pulse, or output tot? also add hit if never over threshold?
+        // Add the hit to the hitmap
+	//        hits.emplace_back(pixel, toa, output_pulse.getCharge(), &pixel_charge);
+        hits.emplace_back(pixel, toa, tot, &pixel_charge);
+    }
+
+    // Output summary and update statistics
```

Comment:
I'm not a huge fan of this sort of statistics, if you think it helps really then leave it in, otherwise remove it I would suggest.